lisa/analysis/rta.py

```diff
@@ -0,0 +1,337 @@
+
+class Events(TraceAnalysisBase):
+    """
+    Support for RTA events analysis.
+
+    :param trace: input Trace object
+    :type trace: :class:`trace.Trace`
+    """
+
+    name = 'rta'
+
+    def __init__(self, trace):
+        """
+        Load peformance data of an rt-app workload
+        """
+        logger = self.get_logger()
+
+        self.trace = trace
+
+        if 'rtapp_task' not in self.trace.base_trace.available_events:
+            raise ValueError('No rt-app tasks events found in trace')
+
+        tasks_df = self.trace.base_trace.df_events('rtapp_task')
+        self.rtapp_tasks = unique(tasks_df["__comm"].values).tolist()
+
+        logger.info("Found rt-app tasks: %s", self.rtapp_tasks)
+
+    def _task_filtered(self, df, task=None):
+        if not task:
+            return df
+        if task not in self.rtapp_tasks:
+            raise ValueError("Task [{}] is not an rt-app task: {}"
+                             .format(task, self.rtapp_tasks))
+        return df[df["__comm"] == task]
+
+
+###############################################################################
+# DataFrame Getter Methods
+###############################################################################
+
+    ###########################################################################
+    # rtapp_main events related methods
+    ###########################################################################
+
+    @requires_events('rtapp_main')
+    def df_rtapp_main(self):
+        """
+        Dataframe of events generated by the rt-app main task.
+
+        :returns: a :class:`pandas.DataFrame` with:
+
+          * A ``__comm`` column: the actual rt-app trace task name
+          * A ``__cpu``  column: the CPU on which the task was running at event
+                                 generation time
+          * A ``__line`` column: the ftrace line numer
+          * A ``__pid``  column: the PID of the task
+          * A ``data``   column: the data corresponding to the reported event
+          * An ``event`` column: the event generated
+
+        The ``event`` column can report these events:
+
+          * ``start``: the start of the rt-app main thread execution
+          * ``end``: the end of the rt-app main thread execution
+          * ``clock_ref``: the time rt-app gets the clock to be used for logfile entries
+
+        The ``data`` column reports:
+
+          * the base timestamp used for logfile generated event for the ``clock_ref`` event
+          * ``NaN`` for all the other events
+
+        """
+        return self.trace.base_trace.df_events('rtapp_main')
+
+    @property
+    @df_rtapp_main.used_events
+    def rtapp_window(self):
+        df = self.df_rtapp_main()
+        return (
+            df[df.event == 'start'].index.values[0],
+            df[df.event == 'end'].index.values[0])
+
+    @property
+    @df_rtapp_main.used_events
+    def rtapp_reftime_map(self):
+        df = self.df_rtapp_main()
+        return {
```

Comment:
Will this dict grow in the future ? If not, a tuple might be better suited

Reply:
Could be, the `rtapp_main` event is generic and it can host other reference times maybe...

Reply:
Updated to return a named tuple...